src/agentconfig.rs

```diff
@@ -201,10 +201,49 @@ scheme of \"productname-productversion\".
         };
         None
     }
+
+    pub fn get_documentation() -> Result<String, AgentConfigError> {
+        let mut doc_string = String::new();
+        for option in AgentConfig::get_options() {
+            doc_string.push_str(&format!("\n\n\n### {}\n\n", option.name));
+            doc_string.push_str(&format!(
+                "*Default value*: {}\n\n",
+                option.default.unwrap_or("No default value")
+            ));
+            doc_string.push_str(&format!("*Required*: {}\n\n", option.required));
+            doc_string.push_str(&format!("*Multiple values:*: {}\n\n\n", option.list));
+
+            // We have not yet specified a documentation string for all options, as an interim
+            // solution we use the help string for the docs, if no proper doc has been written yet.
+            if option.documentation.is_empty() {
+                doc_string.push_str(&option.help);
+            } else {
+                doc_string.push_str(&option.documentation);
+            }
+        }
+        Ok(doc_string)
+    }
+
+    fn gen_docs() {
```

Comment:
As you say yourself: This should be moved to a src/bin/gen_docs.rs file or something along those lines